### DISCUSSION TITLE
Proper parameter ranges in Chen2020_plating

### DISCUSSION THREAD

**Discussion by Mrzhang-hub** [pybamm-team/PyBaMM#2030] [Q&A]
I am doing parameter sensitivity analysis and parameter identification using  LGM50 battery from Chen2020_plating. But I could not find proper parameter ranges for these parameters. Because I need to choose parameters from parameter ranges randomly when identifying parameters, it sometimes goes wrong when the parameter set is chosen improperly.  Following is the key part of my code

```
#parameters chosen for sensitivity analysis and identification
param_list = ['Negative particle radius [m]',  
              'Positive particle radius [m]',  
              'Negative electrode diffusivity [m2.s-1]',  
              'Positive electrode diffusivity [m2.s-1]', 
              'Negative electrode conductivity [S.m-1]', 
              'Positive electrode conductivity [S.m-1]',  
              'Electrolyte conductivity [S.m-1]',  
              'Negative electrode active material volume fraction',  
              'Positive electrode active material volume fraction',  
              'Maximum concentration in negative electrode [mol.m-3]', 
              'Maximum concentration in positive electrode [mol.m-3]', 
              'Initial concentration in electrolyte [mol.m-3]', 
              'Negative electrode Bruggeman coefficient (electrode)',  
              'Positive electrode Bruggeman coefficient (electrode)', 
              'Separator Bruggeman coefficient (electrolyte)']  

#parameter ranges  lb: lower bound  ub: upper bound
lb = [1.86e-6, 1.22e-6, 7e-15, 7.8e-16, 0.5, 0.06, 1, 0.7, 0.615, 29867, 63105, 800, 1.5, 1.5, 1.1]  
ub = [7.86e-6, 7.22e-6, 5.5e-14, 7.8e-15, 400, 15, 2, 0.8, 0.715, 43000, 73000, 1200, 3.4, 3.4, 1.9]

#model 
model = pybamm.lithium_ion.SPMe({
                "lithium plating": "irreversible",
                "lithium plating porosity change": "true",
                "thermal": "lumped"})

#parameters
 parameter_values = pybamm.ParameterValues('Chen2020_plating')

#experiment 
    experiment = pybamm.Experiment(
        [
            "Discharge at 0.5C until 2.55 V (5 seconds period)",
            "Rest for 2 hours",
        ],
        period="30 seconds",
    )
```

When randomly combining a parameter set to run the simulation, it sometimes gives the following mistake:
`Experiment is infeasible: 'event: Maximum voltage' was triggered during 'Discharge at 0.5C until 2.5 V (5 seconds period)'. The returned solution only contains the first 0 cycles. Try reducing the current, shortening the time interval, or reducing the period.`

So is there any paper that gives the proper parameter ranges for LGM50 battery?

**Comment by brosaplanella** [comment]
I believe the issue might be because you change the maximum concentration in the electrodes but don't adjust the initial concentrations accordingly, so you might end up getting initial voltages above the cell cut-off. A way to go would be to determine the initial stoichiometry of the original parameters and, when you update the maximum concentration, update the initial concentrations so the stoichiometries remain the same.

**Reply by Mrzhang-hub** [reply]
Thanks, it works for me. And I wanna ask one more question. 

```
#Access to the negative electrode reaction overpotential
sim.step(0.01,save=False,starting_solution=sim.solution,inputs={'Current function [A]':Batt_I/12})
Op_n = sim.solution['Negative electrode lithium plating reaction overpotential [V]'].entries
```
The result of the negative electrode reaction overpotential is as the following figure. I am confused with that it have twenty rows and two columns.  In my view, the column should stand for overpotential in positive electrode and negative electrode and, the row stands for the electrode position. But here I only wanna know that in  negative electrode which should be one column from my perspective. Can you explain the meaning of this 20*2 matrix for me?  
![Overpotential](https://user-images.githubusercontent.com/59716473/164711958-303b7d7e-fe78-44e3-adde-a2cbdc2d5f73.png)

**Reply by brosaplanella** [reply]
Note that the overpotenital is a spatially distributed quantity, so the columns should be the two time points (before and after the step) and rows should correspond to the different gridpoints along the `x` coordinate in the negative electrode.

**Reply by Mrzhang-hub** [reply]
I got it. So is the time interval between these two time points  a fixed value? If fixed, how long is the time interval?    I found the overpotential is not continuous, maybe caused by that my simulation time interval is different from the time interval between these two points.

**Reply by brosaplanella** [reply]
The time interval should be `0.01` as this is what you have prescribed when calling `step`.
